internal/node/proxy.go

```diff
@@ -138,11 +137,11 @@ func (r TwinResolver) PublicKey(twin int) ([]byte, error) {
 	if ok {
 		return cached.([]byte), nil
 	}
-	pk, err := r.client.GetTwinPK(uint32(twin))
+	twinObj, err := r.client.GetTwin(uint32(twin))
```

Comment:
replace `twin` -> `twinId`, and `twinObj` with `twin`, also change the signature of PublicKey function to use uint32 instead of int for the twin, this will prevent the type conversion and eliminate errors coming from passing the wrong signed values for twin